core/field_variable.js

```diff
@@ -248,9 +285,8 @@ Blockly.FieldVariable.prototype.getVariableTypes_ = function() {
   var variableTypes = this.variableTypes;
   if (variableTypes === null) {
     // If variableTypes is null, return all variable types.
-    if (this.sourceBlock_) {
-      var workspace = this.sourceBlock_.workspace;
-      return workspace.getVariableTypes();
+    if (this.workspace_) {
+      return this.workspace_.getVariableTypes();
     }
```

Comment:
I don't think so. If the field's variableTypes property isn't defined, all variable types are valid, correct? Based on the logic of typeIsAllowed_, getVariableTypes should just return something falsy (like null) if variableTypes isn't defined.

It might be better to investigate this separately, if you agree I can put up an issue for it.

Reply:
Confirmed, separate issue but should investigate before the next release.

Reply:
Added Issue #2483